### PR TITLE
Hook in agg expressions to the table.agg API (global mode only).

### PR DESCRIPTION
**Pull request by xcharleslin**
Implement the table.agg API, but only for global mode. This is so we can start using ungrouped aggregations right away instead of going through eval_expression_list.

### CODE REVIEW

File: src/table/ops/agg.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        match group_by.len() {
+            0 => self.agg_global(to_agg),
+            _ => self.agg_groupby(to_agg, group_by),
+        }
+    }
+
+    pub fn agg_global(&self, to_agg: &[(Expr, &str)]) -> DaftResult<Table> {
+        // Convert the input (child, name) exprs to the enum form.
+        //  e.g. (expr, "sum") to Expr::Agg(AggEXpr::Sum(expr))
+        // (We could do this at the pyo3 layer but I'm not sure why they're strings in the first place)
```
[the remark this answers is not in the document]
Comment (jaychia) [reply src/table/ops/agg.rs:18]: So I did some more digging after you asked about it yesterday

This is an artifact of our old code where we didn't allow for evaluating on aggregation expressions, and the evaluation was handled separately outside of `eval_expressions` as a separate call to `Block.agg` which takes a string argument: https://github.com/Eventual-Inc/Daft/blob/main/daft/runners/partitioning.py#L461-L484

It does seem a little odd though and we can probably change that behavior. But yes in the interest of speed we can just keep this API first until we're fully integrated.

Maybe @samster25 can add some more color here

File: tests/table/test_table.py [hunk 2]
```diff
@@ -439,6 +439,70 @@ def test_table_sum_badtype() -> None:
...
+
+
+@pytest.mark.parametrize("case", test_table_agg_global_cases, ids=[f"{_}" for _ in test_table_agg_global_cases])
+def test_table_agg_global(case) -> None:
+    """Test that global aggregation works at the API layer."""
+    input, expected = case
+    daft_table = Table.from_pydict({"input": input})
+    daft_table = daft_table.agg(
+        [
+            (col("input").cast(DataType.int32()).alias("count"), "count"),
```
[the remark this answers is not in the document]
Comment (jaychia) [reply tests/table/test_table.py:493]: Yes, the type checking logic is entirely in your control, so feel free to make any changes you deem fit. I also did run into issues too with some of the table tests which run on empty arrays and resorted to casting like you do so here.

The one thing we should try to adhere to though is that things that pass type-checks should actually be runnable and vice versa :P 

I can run you through the unit test which I use to do that tomorrow (https://github.com/Eventual-Inc/Daft/blob/main/tests/test_runtime_matches_static_types.py#L70-L71)
